### DISCUSSION TITLE
Swar view fails periodically with "OSError: [Errno 12] Not enough space"

### DISCUSSION THREAD

**Discussion by Scorillo47** [swar/Swar-Chia-Plot-Manager#350] [Q&A]
I have this issue reproducing consistently on all my Windows machines. I am using Windows 10,  and Python 3.7.10 (default, Feb 26 2021, 13:06:18) [MSC v.1916 64 bit (AMD64)] :: Anaconda, Inc. on win32

> Restarting view due to psutil going stale...
> Traceback (most recent call last):
>   File "manager.py", line 36, in <module>
>     view()
>   File "D:\_chia\Swar-Chia-Plot-Manager\plotmanager\library\utilities\commands.py", line 123, in view
>     os.execv(sys.executable, system_args)
> OSError: [Errno 12] Not enough space
> 

Maybe the view process has perhaps a memory leak of some sort? The view gets blocked when this message appears and I have to manually restart it, after which it again works fine for a while.

**Comment by swar** [comment]
This comes down to Windows not allowing more memory to be assigned. No idea how to fix this issue, but I am moving this to Discussions

**Reply by Scorillo47** [reply]
Can you please detail what do you mean? 

Is this a problem with the Python runtime or the Chia implementation in Windows?

**Reply by Scorillo47** [reply]
Found a possible issue. SWAR calls os.execv(sys.executable, system_args) which is problematic in Windows. 
- https://bugs.python.org/issue19124

In fact on Windows it is recommended to use the subprocess class which makes more sense
- https://www.programcreek.com/python/example/986/os.execv
- https://stackoverflow.com/questions/7264571/how-to-do-os-execv-in-python-in-windows-without-detaching-from-the-console 

For Windows, I recommend simply exiting with a certain exit code value (say "1") and execute this in a batch file. Let me try this in my scripts

**Reply by Scorillo47** [reply]
Fixed the problem. I just added an exit(1) after the "stale" detection. Also I am simply executing the view script in a loop. That's it.

**Reply by swar** [reply]
Hey sorry I didn't see this earlier. I've been absolutely overloaded with messages that it's hard to keep up with everything. When you say you run this in a loop, is this outside of the view script? Like a powershell loop or something?

**Reply by Scorillo47** [reply]
that is correct - just created a batch file that invokes this in a loop. 

I had to add an exit() instead of the os.execv() call and that fixed the problem

**Comment by Scorillo47** [comment]
Yes

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________
